Tests/UnitTests/Purchasing/IntroEligibilityCalculatorTests.swift

```diff
@@ -74,67 +73,117 @@ class IntroEligibilityCalculatorTests: TestCase {
         }
 
         expect(self.mockProductsManager.invokedProductsCount) == 1
-        expect(self.mockProductsManager.invokedProductsParameters) == candidateIdentifiers.union(receiptIdentifiers)
+        expect(self.mockProductsManager.invokedProductsParameters) == candidateIdentifiers
+            .union(receipt.activeSubscriptionsProductIdentifiers)
+            .union(receipt.expiredSubscriptionProductIdentifiers)
     }
 
     func testCheckTrialOrIntroDiscountEligibilityGetsCorrectResult() {
-        let receipt = mockReceipt()
-        mockReceiptParser.stubbedParseResult = receipt
-
-        let product1 = MockSK1Product(mockProductIdentifier: "com.revenuecat.product1",
-                                      mockSubscriptionGroupIdentifier: "group1")
-        product1.mockDiscount = MockSKProductDiscount()
-        let product2 = MockSK1Product(mockProductIdentifier: "com.revenuecat.product2",
-                                      mockSubscriptionGroupIdentifier: "group2")
-        product2.mockDiscount = MockSKProductDiscount()
-
-        mockProductsManager.stubbedProductsCompletionResult = .success(
-            Set([product1, product2].map(StoreProduct.init(sk1Product:)))
+        self.testEligibility(
```

Comment:
🧹 I've refactored all these tests to avoid the duplication and to be able to add more cases more easily.